### PR TITLE
Make all non-admin users in the p3 project into Barbican observers.

### PR DESCRIPTION
**Pull request by oneswig**
Promote Mark Boulton to p3 project admin (for appliance creation).

### CODE REVIEW

File: etc/p3-config/p3-config.yml [hunk 1]
```diff
@@ -141,7 +141,7 @@ p3_users:
           30363431336365303231633165313565623361373766333264343432326566653965316439366334
           3137616331653461340a343332343634646638363630333433343639373463303138383866643436
           3162
-    roles: "{{ p3_user_roles }}"
+    roles: "{{ p3_admin_roles }}"
```
Comment (JohnGarbutt) [inline etc/p3-config/p3-config.yml:144]: This seems good.

File: etc/p3-config/p3-config.yml [hunk 2]
```diff
@@ -223,6 +223,7 @@ p3_admin_roles:
 # List of roles to apply to admin users in the p3 project.
 p3_user_roles:
   - heat_stack_owner
+  - observer
```
Comment (JohnGarbutt) [inline etc/p3-config/p3-config.yml:226]: Hmm, what do we want to allow here?

Observer doesn't appear to be able to read secrets, I may be reading that wrongly.

We might want creator role here?
https://github.com/openstack/barbican/blob/master/etc/barbican/policy.json#L4

Having said that, I am all up for trying the lower level of privilege, and seeing how we go with that.

**Review by JohnGarbutt — COMMENTED** [review]
Not sure if observer is what you want here. What was the aim for the users?

File: etc/p3-config/p3-config.yml (hunk 2 above)
Comment (oneswig) [reply etc/p3-config/p3-config.yml:226]: I was basing it on https://docs.openstack.org/barbican/latest/admin/access_control.html, which says:

*observer*
Users with this role are allowed to access to existing resources but are not allowed to upload new secrets or delete existing secrets.

- Looks like it works, but the test case is in Australia and they are finished for the day...

Comment (JohnGarbutt) [reply etc/p3-config/p3-config.yml:226]: Ah, turn out that is correct.

observer -> all_users -> secret_non_private_read -> secret:get

So darn easy to miss a link when reading those policy files.

**Review by JohnGarbutt — APPROVED** [review]
Cool, looks good to me.

**Comment by oneswig** [comment]
Thanks John

**Pull request merged**